service/src/device_method.ts

```diff
@@ -85,6 +84,49 @@ export class DeviceMethod {
     const totalTimeout = (this.params.responseTimeoutInSeconds + this.params.connectTimeoutInSeconds) * 1000;
     this._client.executeApiCall('POST', path, headers, this.params, totalTimeout, done);
   }
+
+
+  /**
+   * @method            module:azure-iothub.DeviceMethod.invokeOnModule
+   * @description       Invokes the method on the specified module with the specified payload.
+   * @param {String}    deviceId    Identifier of the device on which the method will run.
+   * @param {String}    moduleId    Identifier of the module on which the method will run.
+   * @param {Function}  done        The function to call when the operation is
+   *                                complete. done` will be called with three
+   *                                arguments: an Error object (can be null), the
+   *                                body of the response, and a transport-specific
+   *                                response object useful for logging or
+   *                                debugging.
+   */
+  invokeOnModule(deviceId: string, moduleId: string, done: DeviceMethod.ResponseCallback): void {
+    /*Codes_SRS_NODE_IOTHUB_DEVICE_METHOD_18_001: [The `invokeOnModule` method shall throw a `ReferenceError` if `deviceId` or `moduleId` is falsy. ]*/
+    if (!deviceId) throw new ReferenceError('deviceId cannot be \'' + deviceId + '\'');
+    if (!moduleId) throw new ReferenceError('moduleId cannot be \'' + moduleId + '\'');
+
+    const path = '/twins/' + deviceId + '/modules/' + moduleId + '/methods' + endpoint.versionQueryString();
```

Comment:
>+ deviceId  [](start = 27, length = 11)

no need to uri encode?

Reply:
good point. i think both need URI encoding.

---
In reply to: [176867302](https://github.com/Azure/azure-iot-sdk-node/pull/256#discussion_r176867302) [](ancestors = 176867302)